lib/rrd.ml

```diff
@@ -224,8 +227,12 @@ let rra_update rrd proc_pdp_st elapsed_pdp_st pdps =
 
         (* Reinitialise the CDP preparation area *)
         let new_start_pdp_offset = (elapsed_pdp_st - start_pdp_offset) mod rra.rra_pdp_cnt in
-        let cdp_init = cf_init_value rra.rra_cf in
-        Array.iter (fun cdp -> cdp.cdp_unknown_pdps <- 0; cdp.cdp_value <- cdp_init) rra.rra_cdps;
+        Array.iteri (fun i cdp ->
+          let ds = rrd.rrd_dss.(i) in
+          let cdp_init = cf_init_value rra.rra_cf ds in
```

Comment:
`cdp_init` is only used once, and could be written straight to `cdp_value`, but is fine as it is too.